File: 3709-find-special-substring-of-length-k/find-special-substring-of-length-k.cpp
class Solution {
public:
    bool hasSpecialSubstring(string s, int k) {
        int n = s.size();
        int first = 0, last = 0;

        while (first < n) {
            if (s[first] != s[last]) {
                if (first - last == k) {
                    return true;
                }
                last = first;
            }
            first += 1;
        }

        return n - last == k;
    }
};

File: 3709-find-special-substring-of-length-k/find-special-substring-of-length-k.py
class Solution:
    def hasSpecialSubstring(self, s: str, k: int) -> bool:
        n = len(s)
        last = 0

        for first in range(n):
            if s[first] != s[last]:
                if first - last == k:
                    return True
                last = first

        return n - last == k
